src/framework/global/io/buffer.cpp

```diff
@@ -51,6 +51,18 @@ Buffer::Buffer(ByteArray* ba)
     }
 }
 
+Buffer::Buffer(const ByteArray& ba)
+{
+    m_ba = ba;
+    m_ref = &m_ba;
+}
```

Comment:
This constructor can be misleading for two reasons:
1. An extra copy of the data will be made
2. We will only read the data, but will not write it down.

```
ByteArray ba = ...
Buffer buf(ba); <- copy
buf.read(...); <- ok
buf.write(...); <- wrong, the data will not be written to ba, although it seems as if it is expected to be
```

Reply:
updated commit by removing Buffer(const ByteArray& ba) constructor